calendar/index.js

```diff
@@ -0,0 +1,273 @@
+import { 
+    createMonthElement, 
+    createDayButton 
+} from './utils/dom.js'; 
+
+import { 
+    getNext12Months, 
+    formatDate
+} from './utils/dates.js'; 
+
+import { CLASS_NAMES } from './utils/constants.js';  
+
+import { 
+    dispatchSyncEvent, 
+    handleSyncEvent, 
+    saveToSessionStorage, 
+    getFromSessionStorage, 
+    clearSessionData, 
+    toggleVisibility, 
+    highlightButton, 
+    clearHighlights,
+    loadAndInjectCSS
+} from '../utils/components.js';  // Utility functions for general component behavior
+
+class IRNMNCalendar extends HTMLElement {
+    constructor() {
+        super();
+        this.label = this.getAttribute('label') || 'Check-in';
+        this.placeholder = this.getAttribute('placeholder') || 'Select a date';
+        this.name = this.getAttribute('name') || 'irnmn-calendar';  
+        this.openDate = new Date(this.getAttribute('openDate') || Date.now());
+        this.startName = this.getAttribute('checkin-date-name') || 'startDate';
+        this.endName = this.getAttribute('checkout-date-name') || 'endDate';
+        this.weekDays = this.getAttribute('weekdays') ? this.getAttribute('weekdays').split(',') :  false;
+        this.startStorageKey = `irnmn-${this.startName}-${this.name}`;
+        this.endStorageKey = `irnmn-${this.endName}-${this.name}`;
+        this.dateLocale = this.getAttribute('date-locale') || 'en-gb';
+
+        this.state = {
+            checkin: null,
+            checkout: null,
+        };
+
+        this.dayButtons = [];
+    }
+
+    async connectedCallback() {
+        const renderedCss = await loadAndInjectCSS('../calendar/css/calendar.css');
+        this.render(renderedCss);
+        this.loadFromSessionStorage();  // Load from sessionStorage and apply necessary classes
+
+        // Listen for custom events tied to the specific "name" attribute
+        document.addEventListener(`checkin-selected-${this.name}`, (e) => this.syncState(e));
+        document.addEventListener(`checkout-selected-${this.name}`, (e) => this.syncState(e));
+    }
+
+    disconnectedCallback() {
+        document.removeEventListener(`checkin-selected-${this.name}`, this.syncState);
+        document.removeEventListener(`checkout-selected-${this.name}`, this.syncState);
+    }
+
+    render(css) {
+        this.innerHTML = `<style>${css}</style>`;
+        this.renderInputGroup();
+        this.renderCalendarPanel();
+        this.renderHiddenInputs();
+        this.loadMonthButtons();
+    }
+
+    renderInputGroup() {
+        const inputGroup = this.createElementWithClasses('div', [CLASS_NAMES.inputGroup]);
```

Comment:
@samuelsilvapt could we not achieve the same thing just using a template literal?